core/logger/audit/audit_logger.go

```diff
@@ -0,0 +1,230 @@
+package audit
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/pkg/errors"
+)
+
+const bufferCapacity = 2048
+
+type Data = map[string]any
+
+type AuditLogger interface {
+	Audit(eventID EventID, data Data)
+}
+
+type AuditLoggerConfig struct {
+	forwardToUrl   string
+	environment    string
+	jsonWrapperKey string
+	headers        []serviceHeader
+}
+
+func NewAuditLoggerConfig(forwardToUrl string, isDev bool, jsonWrapperKey string, encodedHeaders string) (AuditLoggerConfig, error) {
+	environment := "production"
+	if isDev {
+		environment = "develop"
+	}
+
+	if forwardToUrl == "" {
+		return AuditLoggerConfig{}, errors.Errorf("No forwardToURL provided")
+	}
+
+	// Split and prepare optional service client headers from env variable
+	headers := []serviceHeader{}
+	if encodedHeaders != "" {
+		headerLines := strings.Split(encodedHeaders, "\\")
+		for _, header := range headerLines {
+			keyValue := strings.Split(header, "||")
+			if len(keyValue) != 2 {
+				return AuditLoggerConfig{}, errors.Errorf("Invalid headers provided for the audit logger. Value, single pair split on || required, got: %s", keyValue)
+			}
+			headers = append(headers, serviceHeader{
+				header: keyValue[0],
+				value:  keyValue[1],
+			})
+		}
+	}
+
+	return AuditLoggerConfig{
+		forwardToUrl:   forwardToUrl,
+		environment:    environment,
+		jsonWrapperKey: jsonWrapperKey,
+		headers:        headers,
+	}, nil
+}
+
+type AuditLoggerService struct {
+	logger          logger.Logger
+	enabled         bool
+	serviceURL      string
+	serviceHeaders  []serviceHeader
+	jsonWrapperKey  string
+	environmentName string
+	hostname        string
+	localIP         string
+	loggingChannel  chan WrappedAuditLog
+}
+
+// Configurable headers to include in POST to log service
+type serviceHeader struct {
+	header string
+	value  string
+}
```

Comment:
These should be exported:
```suggestion
type ServiceHeader struct {
	Header string
	Value  string
}
```
I don't think that `*string` is necessary in this case.